### ISSUE TITLE
navigationSDK 1.0 and 1.1 

### ISSUE THREAD

**Issue by JanetKao** [mapbox/mapbox-navigation-ios#2783]
1. Setting Langeage Chinese ,it always speaks English.
routeOptions.includesSteps = true
 routeOptions.includesSpokenInstructions = true
routeOptions.locale = Locale(identifier: "zh-Hans")
2.If I set langeage , unit  can't change KM and MILE.
routeOptions.includesSpokenInstructions = true
routeOptions.includesSteps = true
routeOptions.distanceMeasurementSystem = usesMetric ? .metric : .imperial

**Comment by S2Ler** [comment]
This is probably due to https://github.com/mapbox/mapbox-navigation-ios/blob/19b637deac80f9240dd846e7ccb8b517e88e0fa5/MapboxCoreNavigation/NavigationRouteOptions.swift#L135-L136

I'm trying to find out the reason for this code.

**Comment by 1ec5** [comment]
I don’t think that code is at issue. It only gets called in the initializer of NavigationRouteOptions, but it looks like the code above is setting the measurement system after initialization.

**Comment by 1ec5** [comment]
> Setting Langeage Chinese ,it always speaks English.

The problem is that `zh-Hans` is a written locale (Simplified Chinese) while the Voice API expects a spoken locale of `cmn` (Mandarin) or `yue` (Cantonese). The Directions API doesn’t realize that Simplified Chinese text can be used for Mandarin text-to-speech, so it returns English instructions. mapbox/locale-utils#7 tracks adding a fallback from Mandarin to Chinese to fix this issue on the server side.
